src/main/java/Carrot.java

```diff
@@ -0,0 +1,215 @@
+        Scanner in = new Scanner(System.in);
+
+        // Print Command List
+        printHelpCommand();
+
+        // Variables to hold current userInput and associated command
+        String userInput;
+        CommandType userCommand;
+
+        // Chat input and response Loop
+        while (true) {
+            userInput = in.nextLine().trim();
+            userCommand = CommandManager.getCommandType(userInput);
+
+            switch (userCommand) {
+            case EXIT:
+                return;
+            case INVALID:
+                printInvalidCommand();
+                break;
+            case HELP:
+                printHelpCommand();
+                break;
+            case LIST:
+                printListItems();
+                break;
+            case TODO:
+                addTodoToList(userInput);
+                break;
+            case DEADLINE:
+                addDeadlineToList(userInput);
+                break;
+            case EVENT:
+                addEventToList(userInput);
+                break;
+            case MARK:
+            case UNMARK:
+                handleMarkUnmarkTask(userCommand, userInput);
+                break;
+            default:
+                break;
+            }
+        }
+    }
+
+    private static void addTodoToList(String userInput) {
+        String[] todoCommandArguments =
+                CommandManager.getCommandArguments(CommandType.TODO, userInput);
+
+        String taskDescription =
+                todoCommandArguments[0];
+
+        Todo task = new Todo(taskDescription);
+
+        listOfTasks.add(task);
+        printAddedTask(task);
+    }
+
+    private static void addDeadlineToList(String userInput) {
+        String[] deadlineCommandArguments =
+                CommandManager.getCommandArguments(CommandType.DEADLINE, userInput);
+
+        String taskDescription =
+                deadlineCommandArguments[0];
+        String by =
+                deadlineCommandArguments[1];
+
+        Deadline task = new Deadline(taskDescription, by);
+
+        listOfTasks.add(task);
+        printAddedTask(task);
+    }
+
+    private static void addEventToList(String userInput) {
+        String[] eventCommandArguments =
+                CommandManager.getCommandArguments(CommandType.EVENT, userInput);
+
+        String taskDescription =
+                eventCommandArguments[0];
+        String from =
+                eventCommandArguments[1];
+        String to =
+                eventCommandArguments[2];
+
+        Event task = new Event(taskDescription, from, to);
+
+        listOfTasks.add(task);
+        printAddedTask(task);
+    }
+
+    private static void handleMarkUnmarkTask(CommandType command, String userInput) {
+        boolean isDone = (command == CommandType.MARK);
+
+        String taskIndex =
+                CommandManager.getCommandArguments(command, userInput)[0];
+        int taskIndexInt =
+                Integer.parseInt(taskIndex);
```

Comment:
Can consider moving this up as it is less than 120 characters (more than 1 occurrance)